lib/rules/no-control-regex.js

```diff
@@ -51,29 +51,42 @@ module.exports = {
             return null;
         }
 
+
+        const controlChar = /[\x00-\x1f]/g; // eslint-disable-line no-control-regex
+        const doubleSlashes = /\\\\/g;
+        const multipleSlashes = /\\+$/g;
+
         /**
-         * Check if given regex string has control characters in it
+         * Return a list of the control characters in the given regex string
          * @param {string} regexStr regex as string to check
-         * @returns {boolean} returns true if finds control characters on given string
+         * @returns {array} returns a list of found control characters on given string
          * @private
          */
-        function hasControlCharacters(regexStr) {
+        function getControlCharacters(regexStr) {
 
             // check control characters, if RegExp object used
-            let hasControlChars = /[\x00-\x1f]/.test(regexStr); // eslint-disable-line no-control-regex
+            const controlChars = regexStr.match(controlChar) || [];
+
+            let stringControlChars = [];
 
             // check substr, if regex literal used
             const subStrIndex = regexStr.search(/\\x[01][0-9a-f]/i);
 
-            if (!hasControlChars && subStrIndex > -1) {
+            if (subStrIndex > -1) {
 
                 // is it escaped, check backslash count
-                const possibleEscapeCharacters = regexStr.substr(0, subStrIndex).match(/\\+$/gi);
+                const possibleEscapeCharacters = regexStr.slice(0, subStrIndex).match(multipleSlashes);
 
-                hasControlChars = possibleEscapeCharacters === null || !(possibleEscapeCharacters[0].length % 2);
+                const hasControlChars = possibleEscapeCharacters === null || !(possibleEscapeCharacters[0].length % 2);
+
+                if (hasControlChars) {
+                    stringControlChars = regexStr.slice(subStrIndex, -1).replace(doubleSlashes, ",\\").split(",");
```

Comment:
If a comment is needed, then the code isn't good enough :-) I'll change it to make the code clearer, rather than adding comments ie cruft.